### DISCUSSION TITLE
Current day real time partitioned assets.

### DISCUSSION THREAD

**Discussion by gustavo-delfosim** [dagster-io/dagster#14370] [Q&A] [answered]
Hi!

I need to build to build assets with real time partitions, i.e., 5-min (or similar) time windows. I only care about the materializations on the current day. As such, I defined my partition as:

```py
realtime_partitions_def = TimeWindowPartitionsDefinition(
    cron_schedule="*/5 * * * *",
    start=pd.Timestamp("now", tz=TIMEZONE).normalize().strftime(realtime_fmt),
    timezone=TIMEZONE,
    fmt=realtime_fmt,
)
```

I have some related questions:
- How often will Dagit reload this definition causing the partition `start` to be updated?
- Even if I update the partition `start`, I guess there will be residual data in the dagster DB regarding previous days partitions. How can I clean it up?

**Comment by clairelin135** [comment, marked as answer]
Hi @gustavo-delfosim.

- Dagit will refetch the partition keys every time you reload pages that display partitions content, so after the start of a new day reloading the page should display the partitions for that new day.
- You have a couple options regarding the previous days partitions:
   - Represent the partition key as a string including the date, i.e. define `realtime_fmt = "%Y-%m-%d-%H:%M"` will create partition keys formatted like `'2023-05-22-00:00'`. This ensures uniqueness across days, so there will be no partition key collision across days.
   - If you want your partition key to not include the date I.e. just be `00:00`, there will be partition key collisions across day. You could resolve this by having a schedule that runs daily at midnight that deletes all runs across those partitions. This will delete your logs and materializations, so you will no longer be able to access this history.

**Reply by gustavo-delfosim** [reply]
Hi @clairelin135 ! 

> Dagit will refetch the partition keys every time you reload pages that display partitions content, so after the start of a new day reloading the page should display the partitions for that new day.

That's great. No need to force a code location reload!

> You have a couple options regarding the previous days partitions: (...)

I did not consider the option of having only `HH:MM` as my partition key, but it can actually be pretty useful! Thanks for the idea!

My main concern is the clean-up job. Not only to avoid the partition key collision, but also to free space in Dagster DB. Do you have any code example on how to write such a job? Or maybe just give me some pointers so I can write it!

**Reply by clairelin135** [reply]
Here's an example of how you can do this: https://github.com/dagster-io/dagster/discussions/12047
